File: agent/src/main/java/io/github/lixidong/apiradar/agent/transport/HttpEvent.java
package io.github.lixidong.apiradar.agent.transport;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * 从 agent 上报给 plugin 的 HTTP 调用事件。
 * 与 plugin 端的 HttpEvent 字段保持一致。
 */
public class HttpEvent {
    public String id;
    public long timestamp;
    public String source;        // RestTemplate / OkHttp / ApacheHttpClient ...
    public String method;
    public String url;
    public Map<String, String> requestHeaders = new LinkedHashMap<>();
    public String requestBody;
    public int responseCode;
    public Map<String, String> responseHeaders = new LinkedHashMap<>();
    public String responseBody;
    public long duration;        // ms
    public String error;         // 异常信息（如有）
}

File: plugin/src/main/kotlin/com/lingshi/apiradar/plugin/model/HttpEvent.kt
package com.lingshi.apiradar.plugin.model

/**
 * 与 agent 上报的 JSON 结构一一对应。
 */
data class HttpEvent(
    val id: String = "",
    val timestamp: Long = 0,
    val source: String? = null,
    val method: String? = null,
    val url: String? = null,
    val requestHeaders: Map<String, String> = emptyMap(),
    val requestBody: String? = null,
    val responseCode: Int = 0,
    val responseHeaders: Map<String, String> = emptyMap(),
    val responseBody: String? = null,
    val duration: Long = 0,
    val error: String? = null,
)
